data-prepper-plugins/aws-plugin-api/src/main/java/org/opensearch/dataprepper/aws/api/AwsContext.java

```diff
@@ -0,0 +1,75 @@
+/*
+ * Copyright OpenSearch Contributors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package org.opensearch.dataprepper.aws.api;
+
+import software.amazon.awssdk.auth.credentials.AwsCredentialsProvider;
+import software.amazon.awssdk.auth.credentials.DefaultCredentialsProvider;
+import software.amazon.awssdk.regions.Region;
+import software.amazon.awssdk.services.sts.StsClient;
+import software.amazon.awssdk.services.sts.auth.StsAssumeRoleCredentialsProvider;
+import software.amazon.awssdk.services.sts.model.AssumeRoleRequest;
+
+import java.util.UUID;
+
+public class AwsContext {
```

Comment:
This class seems decoupled from the rest of the AWS plugin.

Maybe the `AwsPlugin` should provide this in a new provider.

```
public void apply(final ExtensionPoints extensionPoints) {
  var extensionsProvider = new AwsExtensionProvider(defaultAwsCredentialsSupplier);
  extensionPoints.addExtensionProvider(new AwsExtensionProvider(defaultAwsCredentialsSupplier));
  // new
  extensionPoints.addExtensionProvider(new AwsContext(extensionsProvider));
}
```

Now, it might look like this:

```
public interface AwsContext {
  AwsCredentialsProvider getOrDefault()
  Region getRegionOrDefault();
}

Reply:
Discussed offline. As an intermediate step to unblock the PR, I made AwsContext an interface and AwsContextImpl a temporary implementation. Once this issue: https://github.com/opensearch-project/data-prepper/issues/2825 gets resolved, AwsContext bean will be provided by aws-plugin.